app.go

```diff
@@ -1,248 +1,226 @@
-    casinoAccountName, level string, signidiceKey *rsa.PrivateKey,
-) {
-    InitLogger(level)
-    if strings.ToLower(level) == "debug" {
-        broker.EnableDebugLogging()
-    }
-    log.Debug().Msg("initializing app")
-
-    app.Router = mux.NewRouter()
-    app.BlockChain.API = eos.New(blockChainUrl)
-    app.BlockChain.ChainID = chainID
-    app.BlockChain.CasinoAccountName = casinoAccountName
-    app.BlockChain.SignidiceKey = signidiceKey
-
-    log.Debug().Msg("Reading private key from wif")
-    if app.BlockChain.KeyBag.Add(wif) != nil {
-        log.Panic().Msg("Malformed private key")
-    }
-    app.BlockChain.API.SetSigner(&app.BlockChain.KeyBag)
-    app.Broker.Url = brokerURL
-    app.Broker.TopicOffsetPath = offsetPath
-    app.Broker.TopicID = topicID
-
-    app.InitializeRoutes()
+type BlockChain struct {
+	ChainID           eos.Checksum256
+	CasinoAccountName string
+	EosPubKeys        PubKeys
+	RSAKey            *rsa.PrivateKey
 }
 
-type BrokerData struct {
-    Digest eos.Checksum256 `json:"digest"`
+type AppConfig struct {
+	Broker     Broker
+	BlockChain BlockChain
+}
+
+type App struct {
+	bcAPI         *eos.API
+	BrokerClient  *broker.EventListener
+	OffsetHandler io.Writer
+	EventMessages <-chan *broker.EventMessage
+	*AppConfig
+}
+
+func NewApp(bcAPI *eos.API, brokerClient *broker.EventListener, eventMessages <-chan *broker.EventMessage,
+	offsetHandler io.Writer,
+	cfg *AppConfig) *App {
+	return &App{bcAPI: bcAPI, BrokerClient: brokerClient, OffsetHandler: offsetHandler,
+		EventMessages: eventMessages, AppConfig: cfg}
 }
 
 func (app *App) processEvent(event *broker.Event) {
-    log.Info().Msgf("Processing event %+v", event)
-    var data BrokerData
-    parseError := json.Unmarshal(event.Data, &data)
-
-    if parseError != nil {
-       log.Warn().Msg("Couldnt get digest from event")
-       return
-    }
-
-    api := app.BlockChain.API
-    signature, signError := rsaSign(data.Digest, app.BlockChain.SignidiceKey)
-
-    if signError != nil {
-       log.Warn().Msg("Couldnt sign signidice_part_2, reason=" + signError.Error())
-       return
-    }
-
-    trx, packedTx, err := GetSigndiceTransaction(api, event.Sender, app.BlockChain.CasinoAccountName, event.RequestID, signature)
-
-    if err != nil {
-        log.Warn().Msg("couldn't form transaction, reason: " + err.Error())
-        return
-    }
-
-    log.Debug().Msgf("%+v", trx)
-
-    result, sendError := api.PushTransaction(packedTx)
-    if sendError != nil {
-        log.Warn().Msg("Failed to send transaction, reason: " + sendError.Error())
-        return
-    }
-    log.Debug().Msg("Successfully signed and sent txn, id: " + result.TransactionID)
+	log.Debug().Msgf("Processing event %+v", event)
+	var data struct {
+		Digest eos.Checksum256 `json:"digest"`
+	}
+	parseError := json.Unmarshal(event.Data, &data)
+
+	if parseError != nil {
+		log.Warn().Msgf("Couldnt get digest from event, reason: %s", parseError.Error())
```

Comment:
I guess, it's error level. Signidice is one of the main functions, and if casino can't  do it he lost money